File: telerik_ui_for_winforms_courseware_chm/Courseware/Appearance/CS/FisheyeBar/WindowsFormsApplication1/RadStackItemsPanelElement.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing.Design;
using System.Text;
using System.Windows.Forms;
using Telerik.WinControls;
using Telerik.WinControls.Layouts;

namespace Telerik.QuickStart.WinControls
{
    public class RadStackItemsPanelElement : RadItem
    {
        private RadItemOwnerCollection items;
        private StackLayoutPanel stackLayout;

        public RadStackItemsPanelElement()
        {
            this.NotifyParentOnMouseInput = false;
            this.ShouldHandleMouseInput = true;
        }

        [RadEditItemsAction]
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public RadItemOwnerCollection Items
        {
            get
            {
                if (this.items == null)
                {
                    this.items = new RadItemOwnerCollection();
                    this.items.Owner = this.StackLayout;
                }

                return items;
            }
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        [TypeConverter(typeof(ExpandableObjectConverter))]
        public StackLayoutPanel StackLayout
        {
            get
            {
                return stackLayout;
            }
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
        [RefreshProperties(RefreshProperties.All)]
        public Orientation StackOrientation
        {
            get
            {
                return this.StackLayout.Orientation;
            }
            set
            {
                this.StackLayout.Orientation = value;
            }
        }

        protected override void CreateChildElements()
        {
            base.CreateChildElements();

            this.stackLayout = new StackLayoutPanel();

            this.Children.Add(this.StackLayout);
        }
    }
}

File: telerik_ui_for_winforms_courseware_chm/Courseware/TPF/VB/FisheyeBar/FisheyeBar/RadStackItemsPanelElement.vb
Imports Telerik.WinControls
Imports Telerik.WinControls.Layouts
Imports Telerik.WinControls.UI
Imports System.ComponentModel
Imports System.Drawing.Design

Public Class RadStackItemsPanelElement
    Inherits RadItem
    Private m_items As RadItemOwnerCollection
    Private m_stackLayout As StackLayoutPanel

    Public Sub New()
        Me.NotifyParentOnMouseInput = False
        Me.ShouldHandleMouseInput = True
    End Sub

    <RadEditItemsAction()> _
    <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
    Public ReadOnly Property Items() As RadItemOwnerCollection
        Get
            If Me.m_items Is Nothing Then
                Me.m_items = New RadItemOwnerCollection()
                Me.m_items.Owner = Me.StackLayout
            End If

            Return m_items
        End Get
    End Property

    <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
    <TypeConverter(GetType(ExpandableObjectConverter))> _
    Public ReadOnly Property StackLayout() As StackLayoutPanel
        Get
            Return m_stackLayout
        End Get
    End Property

    <DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)> _
    <RefreshProperties(RefreshProperties.All)> _
    Public Property StackOrientation() As Orientation
        Get
            Return Me.StackLayout.Orientation
        End Get
        Set(ByVal value As Orientation)
            Me.StackLayout.Orientation = value
        End Set
    End Property

    Protected Overrides Sub CreateChildElements()
        MyBase.CreateChildElements()

        Me.m_stackLayout = New StackLayoutPanel()

        Me.Children.Add(Me.StackLayout)
    End Sub

End Class